pulldown-cmark/src/parse.rs

```diff
@@ -2091,6 +2176,22 @@ impl<'a, F: BrokenLinkCallback<'a>> Iterator for OffsetIter<'a, F> {
     }
 }
 
+fn format_wikilink<'a>(text: &'a str) -> CowStr<'a> {
```

Comment:
I'm not sure it's a good idea to do this in the parser. If I compare this with my two go-to reference implementations:

- commonmark-hs doesn't normalize wikilinks in its parser, as we can see in the [spec file](https://github.com/jgm/commonmark-hs/blob/master/commonmark-extensions/test/wikilinks_title_before_pipe.md). URL percent-encoding happens in the HTML renderer.
- Gollum (the software behind GitHub's Wiki tab) mixes parsing with normalization, but, because of the way it works, it actually [has to consult its database](https://github.com/gollum/gollum-lib/blob/master/lib/gollum-lib/filter/tags.rb#L250), not just string replace. Meaning you would need more elaborate hook handling to do it anyway.

Reply:
Yeah, I think I do prefer normalization of wikilinks out of the parser after giving it some thought. Though I might draw the implementation closest to commonmark-hs and forgo any "batteries included," since how these links specifically are handled tends to be very particular for a user's usecase, to the degree that it'd serve most usecases if the parser simply emitted the link as-is with no transformation (exactly how the third-party wrote it).